### PR TITLE
load user profile

### PR DESCRIPTION
**Pull request by smanini1**
#### Description:

Load user profile information onto the profile view 

---

#### Preview:

ezgif com-video-to-gif (12)

### CODE REVIEW

File: ios-target/ViewModel/Profile/ProfileViewModel.swift [hunk 1]
```diff
@@ -8,6 +8,81 @@
...
+      self?.setUser(user)
+      self?.delegate?.userDidSet()
+      self?.state = .idle
+      }, failure: { [weak self] error in
+        self?.state = .error(error.localizedDescription)
+    })
+  }
+  
+  func hasValidData() -> Bool {
+    return email.isEmailFormatted()
```
Comment (germanStabile) [inline ios-target/ViewModel/Profile/ProfileViewModel.swift:77]: shouldn't you check for firstName, lastName, username presence here as well?

File: ios-target/View/Profile/ProfileViewController.swift [hunk 2]
```diff
@@ -10,9 +10,64 @@ import UIKit
...
+  @IBOutlet weak var lastNameTextField: UITextFieldPadding!
+  @IBOutlet weak var saveProfileButton: UIButton!
+  
   var viewModel = ProfileViewModel()
   
   override func viewDidLoad() {
     super.viewDidLoad()
+    viewModel.delegate = self
+    saveProfileButton.enable(false)
+    viewModel.loadUserProfile()
```
Comment (germanStabile) [inline ios-target/View/Profile/ProfileViewController.swift:25]: why do you need to load user profile here? shouldn't this information be in the session manager already?

**Review by germanStabile — APPROVED** [review]
Looks great! 💪

**Review by mcousillas6 — APPROVED** [review]

File: ios-target/ViewModel/Profile/ProfileViewModel.swift (hunk 1 above)
Comment (smanini1) [reply ios-target/ViewModel/Profile/ProfileViewModel.swift:77]: I dont think they are mandatory, at least not in the sign up

File: ios-target/View/Profile/ProfileViewController.swift (hunk 2 above)
Comment (smanini1) [reply ios-target/View/Profile/ProfileViewController.swift:25]: just checked and there is a currentUser in the dataManager 😅. There is also a url in the api for this so i figured it was meant to be implemented that way 🤷🏽‍♀️

Comment (glm4) [reply ios-target/View/Profile/ProfileViewController.swift:25]: I think it is a good idea to fetch the updated information. It should be a background process though, without blocking anything nor alerting the user.

**Pull request merged**